gradle/scripts/lib/java-publish.gradle

```diff
@@ -81,6 +68,33 @@ configure(projectsWithFlags('publish', 'java')) {
                                 }
                             }
                             shaded.each { dependencies.remove(it) }
+
+                            def resolvedArtifacts = configurations.compileClasspath.resolvedConfiguration.resolvedArtifacts
```

Comment:
> What is a platform dependency?

`platform` dependency is used to constrain dependencies with a POM file.
https://docs.gradle.org/current/userguide/java_platform_plugin.html#sec:java_platform_bom_import
https://github.com/line/armeria/blob/8ee1b980af7699a978e3e5b7030a090f026ff9f3/gradle/scripts/lib/common-dependencies.gradle#L46-L51

> If we use resolvedConfiguration, is this just the resolved configuration?

Right. But compile dependencies will only contain the dependencies specified in `dependencies { ... }` in build.gradle. So transitive dependencies do not affect the resolution result.

> i.e. If two transient dependencies are resolved based on the latest version strategy, would this be used as opposed to a platform dependency?

The original purpose of this PR is to remove L11~L23 because it could set unintended/latest versions than the versions written in depedencies.toml.
When removing the lines, dependencies relying on BOM weren't resolved. This block was added to resolve the dependencies constrained by BOM.